action.go

```diff
@@ -280,6 +191,173 @@ func (c *cmdAction) String() string {
 	return "[" + strings.Join(ss, " ") + "]"
 }
 
+// CmdConfig allows creating Actions for commands using custom logic for
+// determining the ActionProperties without the need for creating a custom
+// Action implementation.
+//
+// This can be useful for working with custom commands provided by Redis
+// modules for which the built in logic may return sub-optimal properties or in
+// case some commands are known to be slow in some cases and therefore should
+// not use connection sharing.
+//
+// All methods on CmdConfig are safe for concurrent use from different
+// goroutines.
+type CmdConfig struct {
+	// ActionProperties is an optional callback that will be called when
+	// creating a new Action using CmdConfig.Cmd or CmdConfig.FlatCmd and is
+	// used to set the ActionProperties for the new Action.
+	//
+	// If ActionProperties is nil, a builtin default callback will be used that
+	// should work with all standard Redis commands but may not return correct
+	// results for custom commands provided by Redis Modules or unreleased
+	// commands.
+	ActionProperties func(cmd string, args ...string) ActionProperties
+}
+
+func defaultActionProperties(cmd string, args ...string) ActionProperties {
```

Comment:
I think it might be useful to make this public, so someone's custom implementation can fallback to it if `cmd` isn't one of the custom commands they're expecting.

Reply:
Done